File: core/src/main/java/io/github/fourlastor/game/demo/round/GameMessage.java
package io.github.fourlastor.game.demo.round;

public enum GameMessage {
    ROUND_START,
    TURN_START,
    TURN_END,
    ABILITY_START,
    NEXT_STEP,
    ABILITY_END;

    public boolean handles(int code) {
        return this.ordinal() == code;
    }
}

File: core/src/main/java/io/github/fourlastor/game/demo/round/GameMessage.kt
package io.github.fourlastor.game.demo.round

enum class GameMessage {
  ROUND_START,
  TURN_START,
  TURN_END,
  ABILITY_START,
  NEXT_STEP,
  ABILITY_END;

  fun handles(code: Int): Boolean {
    return ordinal == code
  }
}
